connectivity/tests/lrp.go

```diff
@@ -0,0 +1,190 @@
+
+			if policy.Spec.SkipRedirectFromBackend != s.skipRedirectFromBackend {
+				continue
+			}
+
+			i := 0
+			lf := check.NewLRPFrontend(policy.Spec.RedirectFrontend)
+			t.NewAction(s, fmt.Sprintf("curl-%d", i), &pod, lf, features.IPFamilyV4).Run(func(a *check.Action) {
+				a.ExecInPod(ctx, ct.CurlCommand(lf, features.IPFamilyV4))
+				i++
+			})
+		}
+	}
+
+	// Tests LRP backend pods to LRP frontend connectivity: traffic gets redirected
+	// based on the configured skipRedirectFromBackend flag.
+	for _, pod := range t.Context().LrpBackendPods() {
+		pod := pod
+
+		for _, policy := range policies {
+			policy := policy
+
+			if policy.Spec.SkipRedirectFromBackend != s.skipRedirectFromBackend {
+				continue
+			}
+
+			i := 0
+			lf := check.NewLRPFrontend(policy.Spec.RedirectFrontend)
+			t.NewAction(s, fmt.Sprintf("curl-%d", i), &pod, lf, features.IPFamilyV4).Run(func(a *check.Action) {
+				a.ExecInPod(ctx, ct.CurlCommand(lf, features.IPFamilyV4))
+
+				if policy.Spec.SkipRedirectFromBackend {
+					a.ValidateFlows(ctx, pod, a.GetEgressRequirements(check.FlowParameters{
+						AltDstIP:   lf.Address(features.IPFamilyV4),
+						AltDstPort: lf.Port(),
+					}))
+				}
+				i++
+			})
+		}
+
+	}
+}
+
+func WaitForLocalRedirectBPFEntries(ctx context.Context, t *check.Test, frontend string, backendsMap map[string][]string) {
+	ct := t.Context()
+	w := wait.NewObserver(ctx, wait.Parameters{Timeout: 20 * time.Second})
+	defer w.Cancel()
+
+	ensureBPFLBEntries := func() error {
+		for _, ciliumPod := range ct.CiliumPods() {
+			node := ciliumPod.Pod.Spec.NodeName
+			backends, ok := backendsMap[node]
+			if !ok {
+				// No LRP backend pods deployed on this node.
+				continue
+			}
+			cmd := strings.Split("cilium bpf lb list -o json", " ")
+			stdout, err := ciliumPod.K8sClient.ExecInPod(ctx, ciliumPod.Pod.Namespace, ciliumPod.Pod.Name, defaults.AgentContainerName, cmd)
+			if err != nil {
+				t.Fatal("Failed to run cilium bpf lb list -o json command:", err)
+			}
+			var resMap map[string][]string
+			err = json.Unmarshal(stdout.Bytes(), &resMap)
+			if err != nil {
+				return fmt.Errorf("error unmarshalling data: %w", err)
+			}
+			// An LB mapping (frontend, backend) for example:
+			// 169.254.169.255:80 (1)   10.244.1.210:8080 (132) (1)
+			lbMap := make(map[string][]string)
+			for frontendEntry, backendEntry := range resMap {
+				// strip the space and parentheses
+				index := strings.Index(frontendEntry, " ")
+				if index > 0 {
+					frontendEntry = frontendEntry[:index]
+				}
+				if len(backendEntry) > 0 {
+					lbMap[frontendEntry] = append(lbMap[frontendEntry], backendEntry...)
+				}
+			}
+			// Check for frontend and backend mapping in the parsed BPF LB map.
+			for _, backend := range backends {
+				found := false
+				for fe, beStr := range lbMap {
+					if fe != frontend {
+						continue
+					}
+					for _, be := range beStr {
+						if strings.Contains(be, backend) {
+							found = true
+							break
```

Comment:
nit; not sure if this is a deal or not but we would continue looping thru the map when be is found. 
Couldn't we just index and loop into the map ie lbMap[frontend]  ok? -> loop?

Reply:
Good catch! Simplified the logic.